site/testsuites/testsuite-javascript/__tests__/tbdex/wallet/quickstart.test.js

```
@@ -0,0 +1,239 @@
+            }
+        }
+        // :snippet-end:
+
+        expect(quote).toBeDefined();
+    });
+
+    test('Create Order', async () => {
+        while (!order) {
+            try {
+                // :snippet-start: walletQuickstartCreateOrder
+                order = Order.create({
+                    metadata: {
+                    from: customerDid.uri,         // Customer's DID
+                    to: pfiDid,                    // PFI's DID
+                    exchangeId: exchangeId,        // Exchange ID from the Quote
+                    protocol: "1.0"                // Version of tbDEX protocol you're using
+                    }
+                });
+                // :snippet-end:
+
+                expect(order).toBeDefined();
+            } catch (e) {
+                // do nothing to let the loop keep running in the event
+                // the exchange hasn't populated yet
+            }
+        }
+    });
+        
+    test('Sign and Submit Order', async () => {
+        expect(async () => {
+            // :snippet-start: walletQuickstartSubmitOrder
+            await order.sign(customerDid);
+            await TbdexHttpClient.submitOrder(order);
+            // :snippet-end:
+        }).not.toThrow();
+    });
+
+    test('Process Close', async () => {
+            // :snippet-start: walletQuickstartProcessClose
+            var close;
+            while (!close) {
+                try {
+                    const exchange = await TbdexHttpClient.getExchange({
+                        pfiDid: pfiDid,
+                        did: customerDid,
+                        exchangeId: exchangeId
+                    })
+
+                    for (const message of exchange) {
+                        if (message instanceof Close) {
+                            close = message
+                        }
+                    }
+                } catch (e) {
+                    // do nothing to let the loop keep running in the event
```

ignore an exception?

---

We are ignoring it because we run into an issue where if we try and immediately fetch the exchange before the PFI has finished populating it, then it will throw a DNE error. To avoid this, we let it catch the error and just try again. This works better than setting a timeout/retry because there's 1) no guarantee that works and 2) if the exchange is available faster than the retry loop then we wasted time waiting.